sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -468,10 +480,37 @@ public Mono<ConfigurationSetting> getConfigurationSetting(ConfigurationSetting s
     public Mono<Response<ConfigurationSetting>> getConfigurationSettingWithResponse(ConfigurationSetting setting,
         OffsetDateTime acceptDateTime, boolean ifChanged) {
         try {
-            return withContext(context
-                -> serviceClient.getConfigurationSettingWithResponseAsync(setting, acceptDateTime, ifChanged, context));
+            // Validate that setting and key is not null. The key is used in the service URL, so it cannot be null.
+            validateSetting(setting);
+            return withContext(
+                context -> serviceClient.getKeyValueWithResponseAsync(
+                    setting.getKey(),
+                    setting.getLabel(),
+                    acceptDateTime == null ? null : acceptDateTime.toString(),
+                    null,
+                    getIfNoneMatchETag(ifChanged, setting),
+                    null,
+                    addTracingNamespace(context))
+                               .onErrorResume(HttpResponseException.class,
+                                   (Function<Throwable, Mono<ResponseBase<GetKeyValueHeaders, KeyValue>>>)
+                                    throwable -> {
+                                        HttpResponseException e = (HttpResponseException) throwable;
+                                        HttpResponse httpResponse = e.getResponse();
+                                        if (httpResponse.getStatusCode() == 304) {
```

Comment:
Will need to run this by @srnagar and @JonathanGiles as this is a runtime behavior change. Now 304 will return null where it was previously throwing an exception/propagating an exception (sync vs async) and this could lead to `NullPointerException`s as I don't believe it's possible for this to return null before. One solution to this would be returning the `ConfigurationSetting` that was passed, but that may not be right as some APIs take partial `ConfigurationSetting`s.

Do the other languages handle a 304 case?

Reply:
Never mind this entire comment, this used to be coded into `ConfigurationClientImpl` which is being removed in this PR in favor of the code generated implementation and this is retaining that functionality.